src/ducks/sep8Send.ts

```diff
@@ -0,0 +1,165 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { Horizon } from "stellar-sdk";
+import { RootState } from "config/store";
+import { settingsSelector } from "ducks/settings";
+import { getErrorMessage } from "helpers/getErrorMessage";
+import { getErrorString } from "helpers/getErrorString";
+import { log } from "helpers/log";
+import { getToml } from "methods/getToml";
+import { submitSEP8PaymentTransaction } from "methods/sep8Send/submitSEP8PaymentTransaction";
+import {
+  ActionStatus,
+  Asset,
+  RejectMessage,
+  Sep8PaymentTransactionParams,
+  Sep8SendInitialState,
+} from "types/types.d";
+
+interface InitiateSep8SendActionResponse {
+  approvalCriteria: string;
+  approvalServer: string;
+  assetCode: string;
+  assetIssuer: string;
+  homeDomain: string;
+  isRegulated: boolean;
+}
+
+export const initiateSep8SendAction = createAsyncThunk<
+  InitiateSep8SendActionResponse,
+  Asset,
+  { rejectValue: RejectMessage; state: RootState }
+>("sep8Send/initiateSep8SendAction", async (asset, { rejectWithValue }) => {
+  const { assetCode, assetIssuer, homeDomain } = asset;
+
+  try {
+    if (!homeDomain) {
+      throw new Error("Something went wrong, home domain is not defined.");
+    }
+
+    const tomlResponse = await getToml(homeDomain);
+
+    for (let i = 0; i < tomlResponse.CURRENCIES?.length; i += 1) {
```

Comment:
`find()` might make it more clear here.

Reply:
Great suggestion! I had forgotten about the `find()` method: https://github.com/stellar/stellar-demo-wallet/pull/155/commits/eb5efa74ee8630ede6e1c50dbef44b600a7c1b03